### ISSUE TITLE
False positive with anonymous class

### ISSUE THREAD

**Issue by ben-manes** [uber/NullAway#102]
It appears that constructing an anonymous instance with a nullable parameter doesn't get inspected correctly.

```
/.../AbstractLinkedDeque.java:346: error: [NullAway] passing @Nullable parameter 'first'
where @NonNull is required
    return new AbstractLinkedIterator(first) {
                                      ^
```

```java
public PeekingIterator<E> iterator() {
  return new AbstractLinkedIterator(first) {
    @Override @Nullable E computeNext() {
      return getNext(cursor);
    }
  };
}

abstract class AbstractLinkedIterator implements PeekingIterator<E> {
  @Nullable E previous;
  @Nullable E cursor;

  AbstractLinkedIterator(@Nullable E start) {
    cursor = start;
  }
  ...
}
```

**Comment by msridhar** [comment]
Reproduced.  Working on a fix

**Comment by msridhar** [comment]
@ben-manes I am hoping to get #104 fixed before cutting our next release, but if a new release with this fix would be helpful to you, let us know

**Comment by ben-manes** [comment]
Thanks. I'm in no rush. I'll just be mindful to remove the suppressions whenever I upgrade, but that's really not a big deal.
